### PR TITLE
Add A Stateful Dev Environment by Jupiter example to gallery

### PR DESCRIPTION
**Pull request by randombet**
## Why are these changes needed?
Jupyter Notebooks offer agents:
stateful code execution
a record of the conversation
an iterative environment where the team can ReAct to data and errors as the project progresses.

Notebooks also allow us humans to examine the team's output in a convenient format and even return to the notebook at a later date, to pick up from where we left off.

Most importantly, this implementation is easy and requires little code

## Checks
Screenshot 2024-01-22 at 11 38 27 PM

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by randombet** [comment]
@sonichi @olimoz @randombet

**Review by ekzhu — DISMISSED** [review]
It’s weird PR description but okay…

File: website/src/data/gallery.json [hunk 1]
```diff
@@ -5,6 +5,12 @@
     "description": "A huggingface space to explore AutoGen group chat, build agents with zero-code, and access source code for reuse.",
     "tags": ["ui"]
   },
+  {
+    "title": "A Stateful Dev Environment by Jupiter",
```
Comment (ekzhu) [inline website/src/data/gallery.json:9]: “Powered by Jupyter Notebook”

File: website/src/data/gallery.json [hunk 2]
```diff
@@ -5,6 +5,12 @@
...
+    "link": "https://github.com/olimoz/AI_Teams_AutoGen/blob/main/JupyterNotebooksForAutoGen.ipynb",
+    "description": "An AutoGen Teams Jupyter Notebook for Agile Code Development",
```
Comment (ekzhu) [inline website/src/data/gallery.json:11]: Powered by Jupyter notebook.

do we need to mention agile here?

**Comment by ekzhu** [comment]
I made some edits to the description. @olimoz waiting for your final resolution :D

**Review by sonichi — APPROVED** [review]

**Comment by olimoz** [comment]
Woops, bit late to this conversation. Please go ahead.

Meanwhile, I have drafted an implementation of the code executor for Jupyter Notebooks.
See: https://github.com/olimoz/AI_Teams_AutoGen/blob/main/JupyterNotebooksForAutoGen_Capability.ipynb

Uses capability, add_to_agent() and a NotebookCodeExecutor which is inspired by IPythonCodeExecutor at
https://app.codecov.io/gh/microsoft/autogen/pull/1405/blob/autogen/coding/ipython_code_executor.py

**Review by olimoz — COMMENTED** [review]
Typo in description, spurious 's' after 'Team'. 
Should read....

"description": "An AutoGen Team Powered by Jupyter Notebook."

**Pull request merged**